openfunctions/README.md

```diff
@@ -145,6 +145,170 @@ output = pipe(prompt)
 print(output)
 ```
 
+## Self-Hosting OpenFunctions
+
+This section provides a guide on how to self-host the OpenFunctions model on your local machine or serve it locally for your enterprise. The server deploys the OpenFunctions-v0 model with uvicorn, while the client interacts with this local server using the OpenAI package (0.28.xx). 
+
+### Setting Up Your Local Server
+
+The server API endpoint mirrors the interface of the API call executed by `openai.ChatCompletion.create`, ensuring compatibility with clients using the OpenAI package.
+
+Ensure you have the required libraries:
+```bash
+!pip install fastapi uvicorn transformers torch
```

Comment:
! is colab-specific?

Reply:
I added the ! to stay consistent with the syntax used in the [OpenFunctions README](https://github.com/ShishirPatil/gorilla/tree/main/openfunctions#introduction), but you're right it's only used in interactive computing environments like colab and wouldn't be used in a regular Python script. I can definitely change this!